lib/hooks/core/hook-http.js

```diff
@@ -35,6 +35,7 @@ function requestWrap(request) {
 
     var localOptions = (typeof options === 'string') ?
       url.parse(options) : options;
+    localOptions.headers = localOptions.headers || {};
```

Comment:
For reference; here's how request deals with this: https://github.com/request/request/blob/master/request.js#L126.